examples/kafka-client-consumer-record-data-binding/kafka_client_consumer_record_data_binding.md

```diff
@@ -1,8 +1,6 @@
 # Kafka client - Consumer record data binding
 
-This shows how to use a `kafka:Consumer` as a simple record consumer. The records from a subscribed topic can be retrieved using the `poll()` function. This consumer uses the builtin byte array deserializer for both the key and the value, which is the default deserializer in the `kafka:Consumer`.
-
-The received records are converted to the user defined type using data-binding.
+The `kafka:Consumer` connects to a given Kafka server, and then receives messages as a `byte[]` or as a user defined type via data-binding. A `kafka:Consumer` is created by giving the server endpoint, group id and the topic. A user-defined type can be created by creating a subtype of `kafka:AnydataConsumerRecord` and overriding the type of `value` field. Once connected, `poll` method is used to receive messages with the intended type from the Kafka server. The received values will be bound to the target type internally. The consumer uses the built-in byte array deserializer for both the key and the value, which is the default deserializer in the `kafka:Consumer`. Use this to receive messages from a Kafka server with the metadata of the messages like `kafka:PartitionOffset` and `timestamp`.
 
```

Comment:
**What does it do?**
The consumer record data-binding allows you to directly bind Kafka messages to subtypes `kafka:AnydataConsumerRecord` .  It does this by using the built-in bytes deserializer for both the key and value. 

**How to use it?**
To use this directly assign the `poll` method’s return value to the declared variable, which is a subtype of `kafka:AnydataConsumerRecord` . A subtype of `kafka:AnydataConsumerRecord` can be created by specifying a user defined type for the `value` field. 

**When to use it?**
Use this to receive messages from a Kafka server with the metadata of the messages like `kafka:PartitionOffset` and `timestamp`. It is important to note that this only works when `kafka:Producer` also uses the built-in bytes  serializer for Ballerina.